### ISSUE TITLE
warning: moving a local object in a return statement prevents copy elision

### ISSUE THREAD

**Issue by kgizdov** [xrootd/xrootd#1262]
```
[ 58%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClFileSystem.cc.o
/build/xrootd4/src/xrootd-4.12.3/src/XrdCl/XrdClUtils.cc: In static member function ‘static std::vector<std::__cxx11::basic_string<char> > XrdCl::Utils::GetSupportedCheckSums(const XrdCl::URL&)’:
/build/xrootd4/src/xrootd-4.12.3/src/XrdCl/XrdClUtils.cc:636:21: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
  636 |     return std::move( ret );
      |            ~~~~~~~~~^~~~~~~
/build/xrootd4/src/xrootd-4.12.3/src/XrdCl/XrdClUtils.cc:636:21: note: remove ‘std::move’ call
[ 58%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClXRootDMsgHandler.cc.o
[ 59%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClMessageUtils.cc.o
[ 59%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClXRootDResponses.cc.o
[ 59%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClFile.cc.o
[ 59%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClFileStateHandler.cc.o
[ 59%] Built target xrdgsiproxy
[ 59%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClCopyProcess.cc.o
[ 60%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClClassicCopyJob.cc.o
[ 60%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClThirdPartyCopyJob.cc.o
[ 60%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClAsyncSocketHandler.cc.o
[ 60%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClForkHandler.cc.o
[ 60%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClChannelHandlerList.cc.o
[ 60%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClCheckSumManager.cc.o
[ 61%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClTransportManager.cc.o
[ 61%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClJobManager.cc.o
[ 62%] Linking CXX executable mpxstats
[ 62%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClFileTimer.cc.o
[ 62%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClPlugInManager.cc.o
[ 62%] Built target mpxstats
[ 62%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClFileSystemUtils.cc.o
[ 63%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClTPFallBackCopyJob.cc.o
[ 63%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClMetalinkRedirector.cc.o
[ 63%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClRedirectorRegistry.cc.o
[ 63%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClZipArchiveReader.cc.o
[ 63%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClXCpCtx.cc.o
[ 63%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClXCpSrc.cc.o
[ 64%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClLocalFileHandler.cc.o
[ 64%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClLocalFileTask.cc.o
[ 64%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClZipListHandler.cc.o
[ 64%] Building CXX object src/XrdCl/CMakeFiles/XrdCl.dir/XrdClOperations.cc.o
/build/xrootd4/src/xrootd-4.12.3/src/XrdCl/XrdClSIDManager.cc: In member function ‘std::shared_ptr<XrdCl::SIDManager> XrdCl::SIDMgrPool::GetSIDMgr(const XrdCl::URL&)’:
/build/xrootd4/src/xrootd-4.12.3/src/XrdCl/XrdClSIDManager.cc:153:21: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
  153 |     return std::move( ptr );
      |            ~~~~~~~~~^~~~~~~
/build/xrootd4/src/xrootd-4.12.3/src/XrdCl/XrdClSIDManager.cc:153:21: note: remove ‘std::move’ call
[ 64%] Linking CXX shared module libXrdCryptossl-4.so
[ 64%] Linking CXX shared library libXrdClient.so
[ 64%] Built target XrdCryptossl-4
In file included from /build/xrootd4/src/xrootd-4.12.3/src/XrdCl/XrdClMetalinkRedirector.cc:8:
/build/xrootd4/src/xrootd-4.12.3/src/XrdCl/XrdClMetalinkRedirector.hh: In member function ‘virtual std::vector<std::__cxx11::basic_string<char> > XrdCl::MetalinkRedirector::GetSupportedCheckSums() const’:
/build/xrootd4/src/xrootd-4.12.3/src/XrdCl/XrdClMetalinkRedirector.hh:94:23: warning: moving a local object in a return statement prevents copy elision [-Wpessimizing-move]
   94 |       return std::move( ret );
      |              ~~~~~~~~~^~~~~~~
/build/xrootd4/src/xrootd-4.12.3/src/XrdCl/XrdClMetalinkRedirector.hh:94:23: note: remove ‘std::move’ call
```

**Comment by xrootd-dev** [comment]
For now I suggest you remove the option to warn about this issue. It's 
very contentious.

**Comment by simonmichal** [comment]
AFAIK this is fixed in 5cc51a08090b874d10180e019e6bceb0372e8445.

**Comment by kgizdov** [comment]
cheers
